docs/concepts/callbacks.md

```diff
@@ -1,2 +1,223 @@
+            index_data='finetuner/tll-test-index-da'
+        ),
+    ]
+)
+```
+
+Below is an example of the metrics as they are printed at the end of fine-tuning:
+
+```bash
+Training [5/5] ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 48/48 0:00:00 0:00:12 • loss: 13.986
+INFO     Done ✨                                                                                          __main__.py:195
+DEBUG    Finetuning took 0 days, 0 hours 3 minutes and 48 seconds                                         __main__.py:197
+INFO     Metric: 'resnet_base_precision_at_k' before fine-tuning:  0.11575 after fine-tuning: 0.53425     __main__.py:210
+INFO     Metric: 'resnet_base_recall_at_k' before fine-tuning:  0.05745 after fine-tuning: 0.27113        __main__.py:210
+INFO     Metric: 'resnet_base_f1_score_at_k' before fine-tuning:  0.07631 after fine-tuning: 0.35788      __main__.py:210
+INFO     Metric: 'resnet_base_hit_at_k' before fine-tuning:  0.82900 after fine-tuning: 0.94100           __main__.py:210
+INFO     Metric: 'resnet_base_average_precision' before fine-tuning:  0.52305 after fine-tuning: 0.79779  __main__.py:210
+INFO     Metric: 'resnet_base_reciprocal_rank' before fine-tuning:  0.64909 after fine-tuning: 0.89224    __main__.py:210
+INFO     Metric: 'resnet_base_dcg_at_k' before fine-tuning:  1.30710 after fine-tuning: 4.52143           __main__.py:210
+INFO     Building the artifact ...                                                                        __main__.py:215
+INFO     Pushing artifact to Jina AI Cloud ...                                                            __main__.py:241
+INFO     Artifact pushed under ID ''                                                                      __main__.py:243
+DEBUG    Artifact size is 83.580 MB                                                                       __main__.py:245
+INFO     Finished 🚀    
+```
+
+For a more detailed explanation of parameters and usage, please refer to the {ref}`evaluation <evaluation>` page.
+
+
+## BestModelCheckpoint
+
+This callback evaluates the performance of the model at the end of each epoch, and keeps a record of the best performing model across all epochs.
+Once fitting is finished the best performing model is saved instead of the most recent model.
+Finetuner determines which model is the best is based on two parameters:
+
+- `monitor`: This parameter is by default `val_loss`, which uses the evaluation data to compare models. Alternatively, you can set this to `train_loss`, which will compare models using the training data. You can specify any metric recorded by the evaluation callback for this parameter.
+- `mode`: Whether the monitored metric should be maximised (`max`) or minimised (`min`). By default the mode is set to `auto`, meaning that it will automatically choose the correct mode depending on the chosen metric: 'min' if the metric is loss and 'max' if the metric is one recorded by the evaluation callback.
```

Comment:
```suggestion
- `mode`: Whether the monitored metric should be maximized (`max`) or minimized (`min`). By default the mode is set to `auto`, meaning that it will automatically choose the correct mode depending on the chosen metric: `min` if the metric is loss and `max` if the metric is one recorded by the evaluation callback.
```